department-of-veterans-affairs/caseflow
# Sivakesh/APPEALS-42522 - Completed Mail Tasks

Resolves [Correspondence and Appeal Tasks: Can view all completed mail tasks](https://jira.devops.va.gov/browse/APPEALS-42522)

# Description
- Added Completed Mail Tasks Section on the Correspondence and Appeal Tasks tab on the Correspondence Details page


## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

## Testing Plan
1. Navigate to correspondence queue (Your Correspondences or Correspondence Cases page)
2. Select a correspondence and notice link (i.e. http://localhost:3000/queue/correspondence/bde5e263-d85b-4e8a-a49d-4a158087fc82/review_package)
3. Remove the review_package (or any other after-pathways) to navigate to the correspondence details page
    1. i.e. **_http://localhost:3000/queue/correspondence/bde5e263-d85b-4e8a-a49d-4a158087fc82_**
4. Verify if Completed Mail Tasks section is added on the Correspondence and Appeal Tasks tab on the Correspondence Details page:
    1. This should be the first section
    2. This is titled "Completed Mail Tasks"
    3. The section should contain any mail task that was marked completed during Intake of this Correspondence
    4. Completed mail tasks should be in a grid
    5. `No previously completed mail tasks prior to intake.` should be shown when there are not any completed mail tasks.
 6. Repeat test steps as necessary for different correspondences.

### Code Climate
Your code does not add any new code climate offenses? If so why?
